### PR TITLE
[ParticleMechanics] Parallelism fix in searching

### PR DESCRIPTION
**Pull request by bodhinandach**
@roigcarlo @loumalouomega 
This fixes the errors that I told you on Monday. It now runs the test 100 times with OMP parallel without any segmentation fault.

### CODE REVIEW

File: applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h [hunk 1]
```diff
@@ -797,38 +798,35 @@ class MPMStrategy
...
 
-                    if (is_found == true)
+                if (is_found == true)
+                {
+                    pelem->Set(ACTIVE);
+                    if (m_GeometryElement == "Triangle")
                     {
-                        pelem->Set(ACTIVE);
-
                         element_itr->GetGeometry()(0) = pelem->GetGeometry()(0);
```
Comment (loumalouomega) [inline applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h]: I don't think this is a efficient way to replace the nodes

Comment (loumalouomega) [reply applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h]: @roigcarlo ?

**Review by loumalouomega — DISMISSED** [review]
Minor comment

File: applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h (hunk 1 above)
Comment (bodhinandach) [reply applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h]: Thanks @loumalouomega. I will talk with @roigcarlo tomorrow!

Comment (loumalouomega) [reply applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h]: element_itr->GetGeometry() = pelem->GetGeometry()?

Comment (roigcarlo) [reply applications/ParticleMechanicsApplication/custom_strategies/strategies/MPM_strategy.h]: Yep you can do that

**Comment by bodhinandach** [comment]
@roigcarlo @loumalouomega as suggested

**Review by loumalouomega — DISMISSED** [review]

**Comment by bodhinandach** [comment]
Sorry, @loumalouomega I cleaned up too much before. And also added a KRATOS_ERROR check. This one is final.

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**